lib/cdo/pardot.rb

```diff
@@ -143,8 +143,8 @@ def self.sync_new_contacts_with_pardot
       # Temporarily limit the accounts synced into Pardot to just those in
       # Teacher role, which is about 25% of total contacts. We will need to
       # purchase additional Pardot capacity to be able import all contacts.
-      where_clause: "pardot_sync_at is NULL AND pardot_id IS NULL AND "\
-                    "Roles like '%Teacher%'",
+      where_clause: "pardot_sync_at IS NULL AND pardot_id IS NULL AND "\
+                    "opted_out IS NULL",
```

Comment:
Change:
- Remove scoping of roles including '%Teacher%'. This was a scoping for initial import to get our most important 25% of contacts imported first. Now we have 100% of contacts imported and can remove this scoping.
- Add "opted_out IS NULL" filter. `opted_out` is true for anyone who has opted out of our Poste email system. Previously, we would import these contacts into Pardot and immediately mark them as do-not email. This would have left us paying for contacts we are not using. With this change, contacts that are opted_out will never even get imported into Pardot. (This only applies to new contacts being considered for insert. Separately, existing Pardot contacts that we update will get `opted_out` set appropriately if it becomes true. So if a contact gets imported into Pardot and later opts out via Poste, we will migrate the opt out setting into Pardot.)